recipes/bison/all/conanfile.py

```diff
@@ -76,6 +81,7 @@ def _configure_autotools(self):
             "--enable-relocatable",
             "--disable-nls",
             "--datarootdir={}".format(os.path.join(self.package_folder, "bin", "share").replace("\\", "/")),
+            "--with-libreadline-prefix={}".format(tools.unix_path(self.deps_cpp_info["readline"].rootpath)),
```

Comment:
`bison` builds fine without the `readline` dependency available.
I can reproduce the problem as soon as I install `readline-devel`.

I propose the following:
- add a `with_readline` option (defaulted to False because it is unneeded).
- add the `--with-libreadline-prefix=...` when `with_readline` is True
- add `gl_cv_lib_readline=no` to the args when `with_readline` is False.

Reply:
Perhaps `gl_cv_lib_readline` can be (ab)used to get crossbuilding work?